### PR TITLE
Handle ``NVMLError_NotSupported`` exception

### PR DESCRIPTION
**Pull request by rjzamora**
Closes https://github.com/rapidsai/ucx-py/issues/982

### CODE REVIEW

File: ucp/__init__.py [hunk 1]
```diff
@@ -64,9 +64,13 @@ def _is_mig_device(handle):
             if _is_mig_device(handle):
                 continue
 
-            total_memory = pynvml.nvmlDeviceGetMemoryInfo(handle).total
-            bar1_total = pynvml.nvmlDeviceGetBAR1MemoryInfo(handle).bar1Total
+            try:
+                bar1_total = pynvml.nvmlDeviceGetBAR1MemoryInfo(handle).bar1Total
+            except pynvml.nvml.NVMLError_NotSupported:
+                # Bar1 access not supported on this device. Skip
+                continue
```
Comment (rjzamora) [inline ucp/__init__.py:71]: @pentschev - Is it "correct" to leave the `large_bar1` element as `False` for devices without Bar1 access?

Comment (pentschev) [inline ucp/__init__.py:71]:
```suggestion
                # Bar1 access not supported on this device, set it to
                # zero (always lower than device memory).
                bar1_total = 0
```

**Review by pentschev — CHANGES_REQUESTED** [review]

**Review by pentschev — APPROVED** [review]
Thanks @rjzamora for working on this PR! 😄

**Pull request merged**